x-pack/plugins/observability_solution/infra/server/lib/helpers/get_infra_metrics_client.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { ESSearchRequest, InferSearchResponseOf } from '@kbn/es-types';
+import type { KibanaRequest } from '@kbn/core/server';
+import type { InfraPluginRequestHandlerContext } from '../../types';
+import { InfraSources } from '../sources';
+import { KibanaFramework } from '../adapters/framework/kibana_framework_adapter';
+
+export type RequiredParams = Omit<ESSearchRequest, 'index'> & {
+  body: {
+    size: number;
+    track_total_hits: boolean | number;
+  };
+};
+
+export type InfraMetricsClient = Awaited<ReturnType<typeof getInfraMetricsClient>>;
+
+export async function getInfraMetricsClient({
+  sourceId,
+  framework,
+  infraSources,
+  requestContext,
+  request,
+}: {
+  sourceId: string;
+  framework: KibanaFramework;
+  infraSources: InfraSources;
+  requestContext: InfraPluginRequestHandlerContext;
+  request?: KibanaRequest;
+}) {
+  const soClient = (await requestContext.core).savedObjects.getClient();
+  const source = await infraSources.getSourceConfiguration(soClient, sourceId);
+
+  return {
+    search<TDocument, TParams extends RequiredParams>(
```

Comment:
What scenario would use the TDocument generic here?  In case we want to override?  Would we need to use the client in that case?

Reply:
I think we should use this client whenever we need to query metrics in infra server. We'd override the `TDocument` if we want ts to infer the type of `hits`